### DISCUSSION TITLE
Fall 2024 Space Teams University - Lunar Search & Rescue Design Challenge - Rules & Info

### DISCUSSION THREAD

**Discussion by connorjak** [SimDynamX/SpaceTeamsPRO#33] [Announcements]
### Changelog:
- `2024/11/26` Special submission form added. Minor tweaks to task wording to better match the real scenarios in the starter code.
- `2024/11/23` Final and Special task due date extended Nov 24 -> Nov 27.
- `2024/11/21` Final task submission form added.
- `2024/11/18` Task 2 and Final completion criteria clarifications.
- `2024/11/16` Task 2 deadline removed due to issues in starter code.
- `2024/11/05` Task 2 deadline changed Nov 10 -> Nov 15.
- `2024/11/01` Added direct link to Task 1 submission form (was previously in email).
- `2024/10/25` Added header for navigating the 3 most-used pages.
- `2024/10/23` Task requirements tweaked. Some links rearranged. Large updates inside competition wiki pages.
- `2024/10/22` Most information for Task Submissions added, Getting Started documentation released, Setup, Learning, and API documentation released.
- `2024/10/21` Edited link for competition Q&A discussion category.
- `2024/10/11` Newly created.

### Please subscribe for notifications on this page! --->

---

| [Rules & Info (Here)](https://github.com/SimDynamX/SpaceTeamsPro/discussions/33) | [Setup & Learning](https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-Competitor-Setup-&-Learning) | [STU API Docs](https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-API-Docs) | [Q&A Discussions Category](https://github.com/SimDynamX/SpaceTeamsPro/discussions/categories/2024-lunar-sar-competition-q-a) | [Starter Code](https://github.com/SimDynamX/STU_Lunar_SAR) | [Starter Code Releases](https://github.com/SimDynamX/STU_Lunar_SAR/releases)
| ----- | ----- | ----- | ----- | ----- | ----- |

# COMPETITION

### This competition has ended; please `Watch` this repository or subscribe to Texas Space Grant Consortium social media for news about the next competition. Link to results comment: https://github.com/SimDynamX/SpaceTeamsPro/discussions/33#discussioncomment-11548639

An incoming lunar lander has crashed and you have precious little time to mobilize all available lunar surface assets to find and rescue the crew. Your Mission – Design the best AI Mission Manager algorithms in Python to conduct Search & Rescue (SAR) operations on the Moon. 

Test and Experience your Mission in a high-fidelity lunar operations simulation using Space Teams PRO.

- University-level design/coding competition
- Teams of 1-10 undergraduate/graduate students
- $10,000 in prizes awarded for the Top Teams

### Scenario
- You begin with a high-fidelity Space Teams simulation that includes a habitat, crew, robots, rovers, and other assets that are already on the lunar south pole near Shackleton Crater performing lunar exploration and mining activities.
- When the lander crashes, the AI Mission Manager that you design/code (python) jumps into action and directs all assets to conduct search and rescue operations. Robots, rovers, and crew have various capabilities, limitations, resource requirements, actuators, and sensors (such as lights, cameras, antennas, lidar, IMUs, wheels, tools) that your Mission Manager can control via commands.
- The terrain is challenging with obstacles, slopes, and permanently shadowed regions.
- Instructions and example code are provided to help you interface your search and rescue code/algorithms with the sensor data and commands within Space Teams.
- Team submissions are scored based on how quickly they can find the crash site and bring a rescue crew to it.

## Important Dates
- October 10th - Applications Open
- October 21st - Applications Close
- October 23rd - Competition Begins
- October 31st 11:59pm CT - Task 1 Submissions Due
- ~~November 15th 11:59pm CT - Task 2 Submissions Due~~ Task 2 due November 24th 11:59pm CT.
- November 27th 11:59pm CT - Final Submissions Due
- November 27th 11:59pm CT - Special Submissions Due
- December 11th - Winners Announced

## Competition Rules
- Team Qualifications: 
  - 1-10 Texas undergraduate and graduate students in STEAM fields
  - Complete team application form on the Texas Space Grant website (https://ig.utexas.edu/texas-space-grant/2024/space-teams-university-lunar-search-rescue-design-challenge/)
  - A designated Team Leader/point of contact
  - A designated Faculty Advisor/point of contact
- Submissions must be made by 11:59pm Central Time on the due dates.
- Team members can be changed with permission, but the max remains 10.
- Judging is administered by the Texas Space Grant Consortium and results are final.
- There are rules on the allowed functions to use in submitted code for a valid submission that does not exploit or work-around the logic of the competition simulations. More details are in the API documentation below.

# SOFTWARE
Space Teams PRO is a fully programmable space system design, simulation and analysis platform for mission planning and operations.  You can build virtually any space mission within Space Teams, as it provides a digital twin of the entire solar system and the physics of the space environment. You can import CAD and analytical models of your own systems, spacecraft, habitats, robots or anything.  For this competition you will be using it to visualize the lunar search and rescue in action, and to test your designs as you optimize your search strategy using various assets in the lunar environment.  
Participants will receive a free account on Space Teams PRO for 3-months.  

More information on the software is provided in the links below.

# LINKS / HELP
### Competition Application (Closed)
- ~~https://ig.utexas.edu/texas-space-grant/2024/space-teams-university-lunar-search-rescue-design-challenge/~~

### Getting Started Document - Everything you need to guide you through the startup process
- https://drive.google.com/file/d/10vDI-6Tq24X1wKsIpurpXJzalxU-iwbD/view?usp=sharing

### Learning Videos
- A full set of introductory video lessons for Space Teams PRO and this competition.  https://www.youtube.com/playlist?list=PLV9n3IHgy_dTQ1Lumj-QYev9qfvlQIndE

- STPro Setup & Learning for Competition https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-Competitor-Setup-&-Learning
  - Starter code distributed from here: https://github.com/SimDynamX/STU_Lunar_SAR (use process in Setup & Learning for cloning this repository)
- Python API Documentation https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-API-Docs

### Account Creation and "Org" (Organization/Team) management, as well as a place to download Space Teams PRO. Make an Org for your competition team.
- https://www.spaceteamspro.com/

### Rules & Info for this competition, with a comments section for questions on rules.
- https://github.com/SimDynamX/SpaceTeamsPro/discussions/33 (this page)

### Discussion Category for other Q&A related to the competition
- [2024 Lunar SAR Competition Q&A discussion category](https://github.com/SimDynamX/SpaceTeamsPro/discussions/categories/2024-lunar-sar-competition-q-a)

### Public-facing website for technical user documentation, discussions, and bug tracking for Space Teams PRO. Come here to learn about the Space Teams platform.
- https://github.com/SimDynamX/SpaceTeamsPRO

### Written technical documentation for general Space Teams PRO usage. Includes installation and setup guides. 
- https://github.com/SimDynamX/SpaceTeamsPro/wiki

### Contact Competition Team - Not for Technical Q&A
- support@SimDynamX.com
- or SpaceTeamsUniversity@tsgc.utexas.edu
- (512) 471-3583

# SUBMISSIONS
Participants will complete two tasks prior to their final submission.  Each task progressively increases in difficulty, helping you refine and implement an AI Mission Manager capable of conducting Search and Rescue (SAR) operations on the Moon. The tasks are outlined as follows:

## Task 1:  One Robot Easy Search
Design/code and submit your AI Mission Manager script that directs one robot/rover to locate the crash site.
- Your objective is to conduct a search pattern to find the crash site.  Once your laser range scanner detects the crashed lander, your program is successful.
- Conditions are 100% daylight, no obstacles, no power limitations, no communication loss, no need for cameras.

| Visibility  | Communication | Resource Management | Robots | Terrain | Obstacles | Completion Criteria |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| Daylight | No loss of signal  | Infinite resources | Only one robot necessary | Crash site is not in shadow | None | Robot Finds Crash Site (moves nearby) |

## Task 2:  One Robot Difficult Search
Improve your Mission Manager code so that you can locate a crash site under more challenging conditions.
- You are still controlling only one robot, but now you have obstacles to avoid, and the crash is within a permanently shadowed region (PSR), so directed lights (headlights, camera lights) are required.  

| Visibility  | Communication | Resource Management | Robots | Terrain | Obstacles | Completion Criteria |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| Limited by lighting | No loss of signal  | Infinite resources | Only one robot necessary | Crash site in shadow | Obstacle hits detected | Robot Finds Crash Site (moves nearby) |

## Final Submission:  Full Scenario Difficult Search
Design the most advanced version of your Mission Manager code to find the crash site in challenging conditions.
- You have multiple robots and crew, each with different capabilities (speed, power capacity, lighting) and limitations (slope, power utilization, ~~oxygen utilization for crew~~).  
- In addition to obstacles you have to be concerned with line-of-sight communication to relocatable antenna relays for data and commanding to/from the base. 
- There is a charging station to charge batteries for assets when needed. 
- In addition to building your strategy to find the crash site, you also need the crew to arrive at the site to rescue survivors and bring them back to the base. 
- LTV1 with at least 50% power remaining must arrive within 10m of crash site.

| Visibility  | Communication | Resource Management | Robots | Terrain | Obstacles | Completion Criteria |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| Limited by lighting | Realistic loss of signal  | Limited resources | All robots necessary | Crash site in shadow | Rocks block motion | Crew/LTV with Required Resources arrives at Crash Site |

## Special Submission: Extend Sim and Impress the Judges (Optional)
In addition to refining the AI Mission Manager, participants have the option to submit a Special/Advanced Submission by making significant innovative extensions to the Lunar Search & Rescue simulation.  This could be related to the technology used to accomplish the challenge, or it could be additional custom designed assets and activities operating in the lunar environment.   This submission is scored based on technical depth of the additional innovations and/or demonstration of novel concepts for space systems or operations in the lunar south pole environment.

In order to qualify for the special submission, your team must have a valid submission for the 1, 2, and Final tasks.

Here are several examples that would constitute a significant extension to the simulation:
- Modified Entity Behavior code for one or more of the robots which is able to persist through loss of signal and path-plan around obstacles to autonomously search
- Designing and model-importing a new robot, which has different performance characteristics that are justified by its design.
- Implementing computer vision, SLAM, or something else that makes productive use of the cameras on the rovers to improve the search
- Advanced Mission Manager code to do optimal control of entities based on autonomy or task-allocation algorithms
- Designing and importing other assets, such as habitats, spacecraft, vehicles, tools or other systems into the simulation along with entity behaviors to demonstrate operations of these additional elements within the simulation.  
- Competitors are free to come up with their own ideas for special submissions

## Task Submission Requirements

Teams will submit their code and supporting assets through GitHub, by submitting a form including a link to the exact commit that represents that submission. Before this, the teams will need to add `Read` permissions (or just as a "Collaborator") for their copy of the STU_Lunar_SAR repository to these GitHub user accounts:
- connorjak
- Cazael
- AlexSein98

The judges will access your repo on GitHub and run the simulation themselves in order to give feedback and do scoring.
- ~~**Task 1 Submission form: https://docs.google.com/forms/d/e/1FAIpQLSdF4w_QgHxagWldRBh9QM-yQMeNoYQZ8MjdKutB4p2hizqF9w/viewform**~~
- ~~**Task 2 Submission form: https://forms.gle/zKqVfhfrXWwKARTo9**~~
- ~~**Final Task Submission form: https://forms.gle/t753sfjGhuMdGuer5**~~
- ~~**Special Submission form: https://forms.gle/Y4XhtthqznVPKTpe8**~~

# JUDGING

| Task | Scored for Competition | Judge Feedback |
| ------------- | ------------- | ------------- |
| 1       | No | Whether the submitted sim works, and whether any function usage is invalid |
| 2       | No | Whether the submitted sim works, and whether any function usage is invalid |
| Final | Yes | If submitted 72+ hours early, will check if submitted sim works, and whether any function usage is invalid |
| Special | (Separate scoring for special competition) | (None) |

- All code and assets that are part of your mission manager & entity behavior scripts should be included in your GitHub repository.
- Git repo commit history will be used as a proof of integrity (many commits are good for collaboration and are a way to show technical depth of work)
  - NOTE: In the SpaceCRAFT app Project Management, the `Push` command is a `git commit` and a `git push`.
  - Let us know if there are circumstances where your submission git repository will not have a normal amount of commits
- If a submission is late, the judges may not give feedback.
- If the final or special submission is late, it will not be scored.

# PRIZES

All prizes are USD amounts of scholarships given out by the Texas Space Grant Consortium and meant to be shared across team members.

Main Competition:
- 1st Prize: $4000
- 2nd Prize: $3000
- 3rd Prize: $2000

Special Competition
- 1st Prize: $1000

---

### Post your questions about these rules below.

**Comment by connorjak** [comment]
https://github.com/SimDynamX/SpaceTeamsPro/releases/tag/v0.27.3 SpaceTeamsPRO v0.27.2 and earlier versions have a bug in the setup process where Git will not install correctly. v0.27.3 has been uploaded and will be deployed to spaceteamspro.com tomorrow (10/24). A workaround for this in the meantime is to install Git for Windows yourself (https://git-scm.com/downloads/win), then proceed with running FirstTimeSetup.bat and the rest of the install steps.

**Comment by connorjak** [comment]
Some wiki pages had slight updates today clarifying what steps are needed in order to have a working Task 1 example sim from the starter code.

**Comment by connorjak** [comment]
Video errata for Lesson 1 and 3 added to the [Setup and Learning page](https://github.com/SimDynamX/SpaceTeamsPro/wiki/STU-Lunar-SAR-Competitor-Setup-&-Learning).

**Comment by connorjak** [comment]
Some tips for Task 1: 
- If you didn't know, the coordinate system visualized in the sim (red/green/blue arrows) lines up exactly with North/West/Up respectively, and red is X, green is Y for the XY coordinates that many of the provided functions use. This is modeled after how real robotic rover navigation is done by JPL and others. Latitiude/Longitude is warped and somewhat unusable for motion planning when near the poles of a planetary body.
- The crash site should be visible in the vicinity of the starting location, in sunlight near a shadowed region. You can use this to guide which direction you choose to start searching, but keep in mind that in future tasks, there is more than one possible place for the crash site to be, and your code is expected to handle any crash site location within given XY bounds.

**Comment by connorjak** [comment]
Also, many important questions have been answered in the Q&A discussions category; please browse through there to ensure you have all the info you need to continue with the competition! https://github.com/SimDynamX/SpaceTeamsPro/discussions/categories/2024-lunar-sar-competition-q-a

**Comment by SkylerHawkins** [comment]
Hello, can you please provide a link to the form that describes how the submission works?

**Reply by erSuyognp** [reply]
https://docs.google.com/forms/d/e/1FAIpQLSdF4w_QgHxagWldRBh9QM-yQMeNoYQZ8MjdKutB4p2hizqF9w/viewform

**Reply by connorjak** [reply]
Yes this one

**Comment by connorjak** [comment]
Thanks everybody for your Task 1 submissions! If your team has not yet submitted, please do so in the next few days. While we do not guarantee a late submission will get feedback, we will try to go through all of the submissions we get to provide feedback so your next submission has a good starting point!

As a reminder, the Task 1 and 2 submissions are not scored for the competition, but if you finish them on time you'll get timely feedback and be on track to finish the Final task.

**Comment by connorjak** [comment]
Task 2 prep is still in progress, expect news on the email we send out soon.

**Comment by connorjak** [comment]
Task 2 released, with cameras inoperable for now. More details and wiki info soon.

STPRO update: 
- Release Notes: https://github.com/SimDynamX/SpaceTeamsPro/releases/tag/v0.27.4
- Download: [spaceteamspro.com](https://www.spaceteamspro.com/)
- Make sure you run FirstTimeSetup.bat again! This must be run for every version update of STPRO.

STU Lunar SAR starter code update:
- Release Notes: https://github.com/SimDynamX/STU_Lunar_SAR/releases/tag/v1.1.0

**Comment by connorjak** [comment]
Revised MoonTrek link with DEM and Slope layers as well as Hillshade:
- https://trek.nasa.gov/moon/#v=0.1&x=-6430.13125146415&y=-14629.127643297626&z=9&p=urn%3Aogc%3Adef%3Acrs%3AIAU2000%3A%3A30120&d=&l=Site01_final_adj_5mpp_ColorHillshade%2Ctrue%2C1&l=Site01_final_adj_5mpp_surf%2Ctrue%2C1&l=Site01_final_adj_5mpp_Slope%2Ctrue%2C1&locale=&b=moon&e=-18411.848857054953%2C-24211.157214830815%2C5551.586354126652%2C-5047.098071764438&sfz=&w=

**Comment by connorjak** [comment]
By the way, there may be some odd behavior in terms of how your python scripts run when showing camera images on-screen using `cv2.imshow()` and/or `cv2.waitKey()`. Try disabling the code that shows your images if you are having weird behavior in your sims.

**Comment by connorjak** [comment]
Found an issue where the Scout rovers on Task 2 would never complete any tasks due to us not including the `OnCommandComplete(Scout1, "MoveToCoord", ...)` function in the example code. We had made a mistake in the code that's supposed to warn you that you haven't created a reaction for a command that finishes, where that warning would never be seen. Will push a fix to STU_Lunar_SAR repo soon.

**Reply by connorjak** [reply]
Fix implemented https://github.com/SimDynamX/STU_Lunar_SAR/releases/tag/v1.2.1

**Reply by erSuyognp** [reply]
I can't see anything moving while I lauch v1.2.1, I saw the MM function there were commands though. I was working with previous release everything was working but now the same code won't. Is there any major updates?

**Reply by connorjak** [reply]
There were changes to many of the files including the example mission manager and entity behavior. All of the changes are required, so as usual the best approach is to copy your current code somewhere else then replace everything with the newest starter code, then add your own code back where needed to keep working on the task

**Reply by connorjak** [reply]
In particular the SetupAllCommands function we added is needed before any commands will work on an entity

**Comment by connorjak** [comment]
Task 2 submission form: https://forms.gle/zKqVfhfrXWwKARTo9

**Comment by connorjak** [comment]
Another starter code fix: https://github.com/SimDynamX/STU_Lunar_SAR/releases/tag/v1.2.2

Please keep working on and submit Task 2 as you are able.

**Comment by connorjak** [comment]
Congrats if you already submitted Task 2.  Some teams ran into issues that had to be fixed in the starter code.  So, if you haven’t already submitted Task 2, see the issue resolution in the starter code repo https://github.com/SimDynamX/STU_Lunar_SAR/releases and in the Q&A https://github.com/SimDynamX/SpaceTeamsPro/discussions, and continue toward your submission.  The Final Task updated Space Teams PRO version and updated starter code will be released on Monday, and you can submit Task 2 whenever you have it working up to the Final task due date.

**Comment by connorjak** [comment]
Final Task released.

STPRO update: 
- Release Notes: https://github.com/SimDynamX/SpaceTeamsPro/releases/tag/v0.27.6
- Download: [spaceteamspro.com](https://www.spaceteamspro.com/)
- Make sure you run FirstTimeSetup.bat again! This must be run for every version update of STPRO.

STU Lunar SAR starter code update:
- Release Notes: https://github.com/SimDynamX/STU_Lunar_SAR/releases/tag/v1.3.0

**Comment by connorjak** [comment]
Script to change out crash site location and obstacle distribution added to starter code. See instructions here: https://github.com/SimDynamX/STU_Lunar_SAR/releases/tag/v1.3.1

**Comment by connorjak** [comment]
I noticed we didn't "announce" this earlier: the API documentation was updated to fit the Final Task API on Tuesday the 19th. 

If you have any questions on the API, starter code format and data structures, or something else, don't hesitate to post it as a question post!

A reminder: if you have an Unreal engine crash, the crash reporter program only sends your crash details to Epic Games, not us! If you get a crash, please report it as an issue and describe what you were doing, especially if you can make steps to consistently reproduce the crash.

**Comment by connorjak** [comment]
Another quick tip: the search area closely resembles the bounds for the obstacle generation (where the rocks start showing up).

**Comment by connorjak** [comment]
Known issue: rover will Complete MoveToCoord commands when it runs out of battery, instead of auto-shutting-down and failing all its commands when about to run out of battery.

**Comment by connorjak** [comment]
Reminder that you can submit multiple times up to the due date, and we'll only score the last submission. We've been giving feedback across all existing submissions and will prioritize early Final task submission feedback so we can let you know if anything is amiss.

**Comment by connorjak** [comment]
Congrats to those who submitted for the Final task! Between now and December 11th we will be scoring submissions. Please keep an eye on your project's GitHub repo and the email account you used to submit; we may need your help to fix minor submission issues that block us from running your code.

Happy Thanksgiving!

**Comment by connorjak** [comment]
Greetings from Texas Space Grant!

First, we want to thank all of the 130 students who participated in our Space Teams University competition this fall for stepping up to the challenge and putting forward their best efforts!
There were 44 student teams representing more than 25 different departments & student organizations from 9 different Texas institutions of higher education: East Texas A&M, El Paso Community College, Texas A&M University, Texas Christian University, Texas Southern University, University of North Texas, University of Texas at Arlington, University of Texas at Austin, and University of Texas at Dallas.

The winning teams for fall 2024 Space Teams University competition are: 
1. AggieSat Laboratory (Texas A&M)
2. HIVE (University of North Texas)
3. Artemis Accords (Texas A&M)

Great job winning teams! The 3-step competition this semester put a new twist on the challenge and provided a good opportunity for top teams to really shine. You should all be proud of your work!

Thanks to everyone for participating and we hope to see you again next semester.
Join our social media and watch for announcements for the spring competition.

Dr. Tim Urban (he/him)
Director, Texas Space Grant Consortium
Research Associate, UT Institute for Geophysics
Jackson School of Geosciences
University of Texas at Austin
10100 Burnet Road
Austin, TX 78758
[Website](https://ig.utexas.edu/tsgc/) | [Facebook](https://www.facebook.com/NASATexasSpaceGrant/) | [Twitter (X)](https://twitter.com/TexasSpaceGrant) | [LinkedIn](https://www.linkedin.com/company/74874047/admin/feed/posts/) | [Instagram](https://www.instagram.com/texasspacegrant/)

**Reply by connorjak** [reply]
As a side note, we received no submissions for the special submission category.
